server/src/main/java/org/elasticsearch/search/fetch/subphase/DocValueFieldsFetchSubPhase.java

```diff
@@ -76,6 +76,7 @@ public void hitsExecute(SearchContext context, SearchHit[] hits) throws IOExcept
 
         hits = hits.clone(); // don't modify the incoming hits
         Arrays.sort(hits, Comparator.comparingInt(SearchHit::docId));
+        List<String> noFormatFields = new ArrayList<>();
```

Comment:
It is a shame to allocate this for *every* search request that needs to fetch in this sub-phase. The escape analysis will not be able to elide the allocation because the list escapes the method in the deprecation logger invocation. Can we collect these differently?

Also, there is a REST test that exposes this deprecation message. Would you update it to include two fields and ensure the format is correct?

Reply:
> It is a shame to allocate this for every search request that needs to fetch in this sub-phase. The escape analysis will not be able to elide the allocation because the list escapes the method in the deprecation logger invocation. Can we collect these differently?

Do you have something particular in mind?

> Also, there is a REST test that exposes this deprecation message. Would you update it to include two fields and ensure the format is correct?

Noted. I didn't see where the test was to start with but no you pointed it out (offline) and the PR CI highlighted it too I'll address it